gcloud-java-bigquery/src/main/java/com/google/gcloud/bigquery/QueryStage.java

```diff
@@ -0,0 +1,431 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.gcloud.bigquery;
+
+import com.google.api.services.bigquery.model.ExplainQueryStage;
+import com.google.api.services.bigquery.model.ExplainQueryStep;
+import com.google.common.base.Function;
+import com.google.common.base.MoreObjects;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.Lists;
+
+import java.io.Serializable;
+import java.util.List;
+import java.util.Objects;
+
+/**
+ * BigQuery provides diagnostic information about a completed query's execution plan (or query plan
+ * for short). The query plan describes a query as a series of stages, with each stage comprising a
+ * number of steps that read from data sources, perform a series of transformations on the input,
+ * and emit an output to a future stage (or the final result). This class contains information on a
+ * query stage.
+ *
+ * @see <a href="https://cloud.google.com/bigquery/query-plan-explanation">Query Plan</a>
+ */
+public class QueryStage implements Serializable {
+
+  static final Function<ExplainQueryStage, QueryStage> FROM_PB_FUNCTION =
+      new Function<ExplainQueryStage, QueryStage>() {
+        @Override
+        public QueryStage apply(ExplainQueryStage pb) {
+          return QueryStage.fromPb(pb);
+        }
+      };
+  static final Function<QueryStage, ExplainQueryStage> TO_PB_FUNCTION =
+      new Function<QueryStage, ExplainQueryStage>() {
+        @Override
+        public ExplainQueryStage apply(QueryStage stage) {
+          return stage.toPb();
+        }
+      };
+  private static final long serialVersionUID = -472281297327952320L;
+
+  /**
+   * Each query stage is made of a number of steps. This class contains information on a query step.
+   *
+   * @see <a href="https://cloud.google.com/bigquery/query-plan-explanation#steps_metadata">Steps
+   *     Metadata</a>
+   */
+  public static class QueryStep implements Serializable {
+
+    static final Function<ExplainQueryStep, QueryStep> FROM_PB_FUNCTION =
+        new Function<ExplainQueryStep, QueryStep>() {
+          @Override
+          public QueryStep apply(ExplainQueryStep pb) {
+            return QueryStep.fromPb(pb);
+          }
+        };
+    static final Function<QueryStep, ExplainQueryStep> TO_PB_FUNCTION =
+        new Function<QueryStep, ExplainQueryStep>() {
+          @Override
+          public ExplainQueryStep apply(QueryStep stage) {
+            return stage.toPb();
+          }
+        };
+    private static final long serialVersionUID = 8663444604771794411L;
+
+    private final String kind;
+    private final List<String> substeps;
+
+    QueryStep(String kind, List<String> substeps) {
```

Comment:
My guess is that that example is not well-formed. In the [Job Resource Docs](https://cloud.google.com/bigquery/docs/reference/v2/jobs#statistics.query.queryPlan.steps) there is no mention of `field.description`